zebra-state/src/service/finalized_state/disk_format.rs

```diff
@@ -249,6 +251,20 @@ impl IntoDisk for orchard::tree::Root {
     }
 }
 
+impl IntoDisk for ValueBalance<NonNegative> {
```

Comment:
I think this (and also `FromDisk`) has to be `NegativeAllowed`.

This is what we are storing in disk. According to the consensus rules the shielded value pools(sprout, sapling, orchard) can never became negative but there is no such a rule for the transparent value pool. If the transparent value pool can be negative then the entire `ValueBalance` needs to be of `NegativeAllowed`. This also means we need to check the shielded value pools rules explicitly.

Reply:
Hang on, let's double-check the spec and the design, before we make any changes.

### Spec

The goal of value pools is to:
> restrict the creation of new money to a specific number of coins in each coinbase transaction

https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/rfcs/0012-value-pools.md#motivation

So there must be consensus rules which make sure that the transparent value pool must be non-negative.  But some of them aren't written in the spec. Instead, it just says:
> Transfers of transparent value work essentially as in Bitcoin

https://zips.z.cash/protocol/protocol.pdf#overview

The design says that the transparent consensus rule exists, but it's not particularly clear.

So I added a more detailed explanation to the design:
https://github.com/ZcashFoundation/zebra/pull/2555/files#r680644500

### Design, Code & Tests

If we're seeing a negative transparent chain value pool balance, that means either:
* someone has created transparent coins outside a coinbase transaction, or
* there is a bug in Zebra's design, code, or tests.

Let's assume there's a bug in Zebra, and treat this like a bug report.

What results are you seeing?
What code are you using to get those results?
What do you expect to see instead?

I did some checking, and I think we've got a sign wrong somewhere.

What if we swapped all the transparent signs around?
I did some updates to the design that should work:
https://github.com/ZcashFoundation/zebra/pull/2555/files#r680640526